################################################################################
SHORT DESCRIPTION: 
################################################################################
Enables QEMU stubdom support

################################################################################
LONG DESCRIPTION: 
################################################################################
The second piece of code that was added was a switch to turn off the backend 
drivers. Since we are in a stubdomain, we do not need to turn on the backend
drivers because dom0 is doing that fo us. 

Note that this code was taken from the following blog post:

https://blog.xenproject.org/2012/12/12/linux-stub-domain/

Basically, all we did was grab the source code, which contains a link to the 
following git repo:

git clone git://xenbits.xen.org/people/aperard/qemu-dm.git
git checkout stubdom-preview1

From there, you can do a git diff to see the changes that were made to setup 
QEMU in a stubdomain. Since we use BitBake to compile QEMU for both dom0 and 
the stubdomain, we do not need to include the extra complexity to turn this
on / off, so the patch is pretty simple. Basically, we just need to tell the
hypervisor that this guest is a stubdomain, and then we compile out the part 
that starts up the backend drivers since this is done for us by dom0. 

################################################################################
CHANGELOG 
################################################################################
Intial Commit: Rian Quinn, quinnr@ainfosec.com, 3/16/2015
Port 2.6.2: Ross Philipson, philipsonr@ainfosec.com, 10/13/2016
Port 4.1: Remove HVM_PARAM_DM_DOMAIN since xen_get_ioreq_server_info()
          makes it unnecessary.

################################################################################
REMOVAL 
################################################################################
This patch can be removed from QEMU once it fully supports being executed in a
stubdomain on it's own. Likely for that to happen, this code will need to be 
included in QEMU upstream. Once that occurs, this patch can be removed from 
out patch queue. 

################################################################################
UPSTREAM PLAN 
################################################################################
We should attempt to upstream this patch ASAP. To do that however, we will need
to add in the logic to enable / disable stubdomain support. We don't do that 
here as it greatly simplifies the patch, which currently we have to maintain. 
However, once we are ready to upstream, you can basically go to the repo
provided above, and most of the needed logic is there to turn on / off 
stubdomain support. 

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None

################################################################################
PATCHES 
################################################################################
--- a/hw/i386/xen/xen-hvm.c
+++ b/hw/i386/xen/xen-hvm.c
@@ -1495,7 +1502,11 @@ void xen_hvm_init(PCMachineState *pcms,
         error_report("xen backend core setup failed");
         goto err;
     }
+
+    /* Since this is in a stubdom, there is no need to setup backend devices */
+#if 0
     xen_be_register_common();
+#endif
 
     QLIST_INIT(&xen_physmap);
     xen_read_physmap(state);
